compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1547,6 +1547,20 @@ impl<'a> Parser<'a> {
         self.expect(&token::Not)?; // `!`
 
         let ident = self.parse_ident()?;
+
+        if self.eat(&token::Not) {
+            // Handle macro_rules! foo!
+            let span = self.prev_token.span;
+            self.struct_span_err(self.prev_token.span, "macro names arn't followed by a `!`")
+                .span_suggestion_verbose(
```

Comment:
I think this one is ok to be inline: we're asking them to remove the whole thing in that is underlined, it should be fine.

```suggestion
                .span_suggestion(
```